File: models/album.go
package models

import "gorm.io/gorm"

// Album represents an album record in the database
type Album struct {
	gorm.Model
	Title  string  `json:"title" binding:"required"`
	Artist string  `json:"artist" binding:"required"`
	Price  float64 `json:"price" binding:"required"`
}

File: frontend/types/album.ts
export interface Album {
  id: number;
  title: string;
  artist: string;
  price: number;
  created_at: string;
  updated_at: string;
  deleted_at?: string | null;
}

export interface CreateAlbumInput {
  title: string;
  artist: string;
  price: number;
}

export interface UpdateAlbumInput {
  title: string;
  artist: string;
  price: number;
}
